src/ChurchCRM/Service/DashboardService.php

```diff
@@ -38,42 +18,78 @@ public function getFamilyCount(): array
         return ['familyCount' => $familyCount];
     }
 
-    public function getPersonCount(): array
+    public function getDashboardStats(): array
     {
-        $personCount = PersonQuery::Create('per')
+        $sInactiveClassificationIds = SystemConfig::getValue('sInactiveClassification');
+        if ($sInactiveClassificationIds === '') {
+            $sInactiveClassificationIds = '-1';
+        }
+        $aInactiveClassificationIds = explode(',', $sInactiveClassificationIds);
+        $aDirRoleChild = explode(',', SystemConfig::getValue('sDirRoleChild'));
+
+        $personCount = 0;
+        $ageStats = [];
+        $classificationStats = [];
+        $genderStats = [];
+        $familyRoleStats = [];
+
+        $people = PersonQuery::Create('per')
+            ->filterByClsId($aInactiveClassificationIds, Criteria::NOT_IN)
             ->useFamilyQuery('fam', 'left join')
             ->filterByDateDeactivated(null)
-            ->endUse()
-            ->count();
+            ->endUse();
 
-        return ['personCount' => $personCount];
-    }
+        foreach ($people as $person) {
+            $personCount++;
 
-    /**
-     * @return array<string, string>
-     */
-    public function getPersonStats(): array
-    {
-        $data = [];
-        $sSQL = <<<SQL
-SELECT
-    lst_OptionName as Classification,
-    count(*) as count
-FROM person_per
-INNER JOIN list_lst ON  per_cls_ID = lst_OptionID
-LEFT JOIN family_fam ON family_fam.fam_ID = person_per.per_fam_ID
-WHERE
-    lst_ID = 1 AND
-    family_fam.fam_DateDeactivated IS NULL
-GROUP BY per_cls_ID, lst_OptionName
-ORDER BY count DESC;
-SQL;
-        $rsClassification = RunQuery($sSQL);
-        while ($row = mysqli_fetch_array($rsClassification)) {
-            $data[$row['Classification']] = $row['count'];
+            $classification = $person->getClassificationName();
+            if (!array_key_exists($classification, $classificationStats)) {
+                $classificationStats[$classification]['count'] = 0;
+                $classificationStats[$classification]['id'] = $person->getClsId();
+            }
+            $classificationStats[$classification]['count']++;
+
+            $gender = $person->getGender();
+            if ($gender != 0 && in_array($person->getFmrId(), $aDirRoleChild)) {
```

Comment:
can we do strict checking here?

```suggestion
            if ($gender !== 0 && in_array($person->getFmrId(), $aDirRoleChild)) {
```

Reply:
I didn't know that php has this feature, will update it as you recommended.

Reply:
Done